sim/src/test/scala/midasexamples/TutorialSuite.scala

```diff
@@ -377,6 +377,23 @@ class TerminationF1Test extends TutorialSuite("TerminationModule") {
   1 to 10 foreach {x => runTest(backendSimulator, args = Seq("+termination-bridge-tick-rate=10", s"+seed=${x}"), shouldPass = true)}
 }
 
+class CustomConstraintsF1Test extends TutorialSuite("CustomConstraints") {
+  def readLines(filename: String): List[String] = {
+    val file = new File(genDir, s"/${filename}")
+    Source.fromFile(file).getLines.toList
+  }
+  it should s"generate synthesis XDC file" in {
+    val xdc = readLines("FireSim-generated.synthesis.xdc")
+    xdc should contain("constrain_synth1")
+    xdc should contain("constrain_synth2 [reg firesim_top/top/sim/target/CustomConstraints_/dut/r0]")
+  }
+  it should s"generate implementation XDC file" in {
+    val xdc = readLines("FireSim-generated.implementation.xdc")
+    xdc should contain("constrain_impl1")
+    xdc should contain("constrain_impl2 [reg WRAPPER_INST/CL/firesim_top/top/sim/target/CustomConstraints_/dut/r1]")
+  }
+}
+
 // Suite Collections
 class ChiselExampleDesigns extends Suites(
```

Comment:
You should add your constraint test to one of the CI groups below.

Reply:
Done